### ISSUE TITLE
./configure: line 28919: syntax error: unexpected redirection

### ISSUE THREAD

**Issue by RattyDAVE** [SDL-Hercules-390/hyperion#175]
I am getting an error on running configure. 

I am using ALPINE 

To assist in debugging here is what I am running. 

```
apk --no-cache add --virtual build-dependencies git perl autoconf automake flex gawk gcc m4 make g++
mkdir /usr/src
cd /usr/src
git clone https://github.com/SDL-Hercules-390/hyperion.git
cd /usr/src/hyperion
./autogen.sh
```

Then I come to

```
./configure --enable-ipv6=no
```

And it fails with

```
---- CUT ----
checking for _FILE_OFFSET_BITS value needed for large files... no
checking for off_t... yes
checking for _LARGEFILE_SOURCE value needed for large files... no
checking whether getopt wrapper kludge is necessary... no
checking for what optimization flags to use ...  -g -g3 -ggdb3 -O3 -march=native -minline-stringops-dynamically -fomit-frame-pointer
configure:
configure:  Package destination directory prefixes:
configure:
configure:     Libraries:  $(libdir)/$(PACKAGE)
configure:     Data:       $(datadir)/hercules
configure:     Locale:     ${datarootdir}/locale
configure:
./configure: line 28919: syntax error: unexpected redirection
```

Any assistance would be good.

Thanks 

DAVE

**Comment by Fish-Git** [comment]
> ```
>  ./configure: line 28919: syntax error: unexpected redirection
> ```

I have _never_ seen this error before.

_**Ever**_.

> I am using ALPINE

I suspect that may be your problem. Have you tried it _without_ using this "ALPINE" thingy?

Am closing this issue as unknown/invalid since there's not enough information provided to prove this is actually a Hercules bug. Rather, I highly suspect it's actually an "ALPINE" and/or user bug.

If you can provide more conclusive evidence that this is actually a Hercules bug, then feel free to re-open this issue at that time.

**Comment by RattyDAVE** [comment]
Alpine is a major distribution. 

Other scripts work fine. And also as it got quite far down in the configure script.

What do you need?

**Comment by Fish-Git** [comment]
> Alpine is a major distribution.

You mean Alpine Linux? Ah. Okay. I'm not much of a Linux person. Sorry about that. I thought it was some add-on product or something. I didn't know it was a Linux distribution. My bad.

> What do you need?

Output from `util/bldlvlck` would be a good start.

A copy (in .zip format please!) of your generated ./configure script would be very helpful too. (so I can see the actual statement complaining about, i.e. line 28919)

Maybe one of the tools or some part of the autoconfiguration process (i.e. see autogen.sh; it's what _creates_ the ./configure script) has a bug in whatever version you're using? (*)

(*) Just a guess! All I'm saying is in all my years as a Hercules developer I've _never_ seen the ./configure script throw such an error!)

**Comment by Fish-Git** [comment]
Temporarily re-opening while issue is researched...

**Comment by Fish-Git** [comment]
Ah HA! Okay. That makes perfect sense. Thank you, @RattyDAVE!

This actually _**is**_ a **bug** in our `configure` script then, brought about by a "[here string](https://www.tldp.org/LDP/abs/html/x17837.html)" that was added to our `configure.ac` "script":

https://github.com/SDL-Hercules-390/hyperion/blob/8c38cc831461a6edce1991158859383ba4170cf6/configure.ac#L3070-L3109

This bug was unfortunately caused by my own inexperience with Linux and autoconf. I'll have to try and see what I can do to get it fixed for you.

I'm not sure quite sure how to actually go about fixing it at the moment, but I'm sure I'll figure something out with or without some help from my fellow team members who are more experience with such things. So hang loose, Dave: it _will_ get fixed sooner or later! Thank you for reporting it!

**Comment by RattyDAVE** [comment]
Thats ok. I can manually change for now. 

I think one of the problems was that an assumption that bash was the default shell. 

This also works

```
/bin/bash ./configure
```

**Comment by Fish-Git** [comment]
Understood! But technically it is a _bug_ and needs to be fixed. One shouldn't have to do that. You should be able to do things in the normal way and it should work.

(And yes, I'm so used to using bash for everything I tend to forget that code in `configure.ac` isn't `!# bin/bash` but rather `#! /bin/sh`. It's an easy thing to forget for someone who isn't a Linux person like me!)

**Comment by Fish-Git** [comment]
Fixed by commits 498e080880944037131b922dfc02a7f7051895a6 and 74830f69ad783278e788703f27067e6e241d9777.

**Comment by vbcoen** [comment]
Redirection should be with only 2 less thans (<<) on line 3108.

Script should start with :+1: 
#!/bin/bash

Personally I have never used any Linux distro without bash installed although some have others also.

Any user having a distro with it missing can simply install it as a part of pre-requisites for the build.

**Comment by Fish-Git** [comment]
> Personally I have never used any Linux distro without bash ...

Neither have I.

But then I haven't used a whole heck of a lot of them either, so that hardly counts for much!&nbsp;`;-)`

> Any user having a distro with it missing can simply install it as a part of pre-requisites for the build.

They shouldn't have to!  That was the whole point!

But it's a moot issue anyway. The problem has already been fixed by the [two previously mentioned commits](https://github.com/SDL-Hercules-390/hyperion/issues/175#issuecomment-464337356).

**Comment by Fish-Git** [comment]
I've re-opened this issue due to having reverted the commits that attempted to fix the problem that actually only ended up making the situation _worse_ by completely breaking the `LIBRARY_PATH`  variable parsing logic.

So we're back to square one.&nbsp;`:(`

We need to find a portable non-bash default shell way of parsing the `LIBRARY_PATH` variable, and since I don't know how to do it _(and apparently neither does the person I got the "fix" from!), **I need some HELP with this one folks!**_

**Comment by Fish-Git** [comment]
Fixed by commit 89f7cadd90df93d18f5099b6f624d82617b7d7ce!

Problem was the single quotes around the <<'END', which was suppressing variable expansion. Removing them so it's just <<END fixes the problem allowing it to work as expected. Thank you Kevin Monceaux!&nbsp;`:)`
